### PR TITLE
upgrade deps

### PR DESCRIPTION
**Pull request by ry**
update some dependencies

### CODE REVIEW

File: cli/Cargo.toml [hunk 1]
```diff
@@ -94,21 +94,21 @@ ring = "=0.16.20"
 rustyline = { version = "=10.0.0", default-features = false, features = ["custom-bindings"] }
 rustyline-derive = "=0.7.0"
 secure_tempfile = { version = "=3.3.0", package = "tempfile" } # different name to discourage use in tests
-semver = "=1.0.13"
+semver = "1.0"
```
Comment (bartlomieju) [inline cli/Cargo.toml:97]: Deps in this file should be pinned

**Review by bartlomieju — CHANGES_REQUESTED** [review]

File: cli/Cargo.toml (hunk 1 above)
Comment (ry) [reply cli/Cargo.toml]: fixed

File: cli/Cargo.toml [hunk 2]
```diff
@@ -120,12 +120,12 @@ deno_bench_util = { version = "0.63.0", path = "../bench_util" }
...
-once_cell = "=1.12.0"
+once_cell = "1.14"
 os_pipe = "=1.0.1"
-pretty_assertions = "=1.2.1"
+pretty_assertions = "1.3"
 test_util = { path = "../test_util" }
-trust-dns-client = "=0.21.2"
-trust-dns-server = "=0.21.2"
+trust-dns-client = "0.22"
+trust-dns-server = "0.22"
```
Comment (bartlomieju) [inline cli/Cargo.toml:128]: These are still not pinned

File: cli/file_watcher.rs [hunk 3]
```diff
@@ -344,8 +343,8 @@ where
 fn new_watcher(
   sender: Arc<mpsc::UnboundedSender<Vec<PathBuf>>>,
 ) -> Result<RecommendedWatcher, AnyError> {
-  let mut watcher: RecommendedWatcher =
-    Watcher::new(move |res: Result<NotifyEvent, NotifyError>| {
+  let watcher = Watcher::new(
```
Comment (bartlomieju) [inline cli/file_watcher.rs:346]: I'm a bit weary about this change... I don't remember exact reason we didn't want to update watcher crate now, so let's see if any problem surface

**Review by bartlomieju — APPROVED** [review]
LGTM, but please pin Windows deps

**Pull request merged**